treeFormatting.js

```diff
@@ -33,14 +33,95 @@ function parenthesizeTree(tree, options){
 	}
 
 	function processNode(node){
+		console.log(node);
 		var nonTerminal = (node.children instanceof Array) && node.children.length;
+		console.log(nonTerminal);
+
 		if (showNewCats && !parens.hasOwnProperty(node.cat)){
+			//console.log(parens);
 			parens[node.cat] = ["["+node.cat+" ", "]"];
 		}
+		// for (item in node){
+		// 	if (item == "id" || item == "cat" || item == "children"){
+		// 		continue;
+		// 	}
+		// 	if (node[item]){
+		// 		if (toneTree.length == 0){
+		// 			toneTree.push(item);
+		// 		}
+		// 		else{
+		// 			toneTree[0] += '.' + item;
+		// 		}
+		// 	}
+		// 	// console.log(item);
+		// 	// console.log(node[item]);
+		// }
 		var visible = invisCats.indexOf(node.cat) === -1 && parens.hasOwnProperty(node.cat);
+		console.log(visible);
+		// if (nonTerminal) {
+		// 	//console.log(parTree);
+		// 	if (visible) {
+		// 		console.log(showTones);
+		// 		if(showTones){
+		// 			// console.log("passed");
+		// 			// console.log(node["func"]);
+		// 			// console.log("node");
+		// 			//console.log(node);
+		// 			if (node["func"] && node["silentHead"]){
+		// 				console.log("both");
+		// 				parTree.push(parens[node.cat][0] + " f.sh");
+		// 			}
+		// 			else if (node["func"]){
+		// 				console.log("func");
+		// 				parTree.push(parens[node.cat][0] + " f");
+		// 			}
+		// 			else if (node["silentHead"]){
+		// 				console.log("sh");
+		// 				parTree.push(parens[node.cat][0] + " sh");
+		// 			}
+		// 			else{
+		// 				console.log("none");
+		// 				parTree.push(parens[node.cat][0]);//pushes the right parens
+		// 			}
+				
+					
+		// 		}
+		// 		else{
+		// 			if (node["func"] && node["silentHead"]){
+		// 				console.log("both");
+		// 				parTree.push(parens[node.cat][0] + " f.sh");
+		// 			}
+		// 			else if (node["func"]){
+		// 				console.log("func");
+		// 				parTree.push(parens[node.cat][0] + " f");
+		// 			}
+		// 			else if (node["silentHead"]){
+		// 				console.log("sh");
+		// 				parTree.push(parens[node.cat][0] + " sh");
+		// 			}
+		// 			else{
+		// 				console.log("none");
+		// 				parTree.push(parens[node.cat][0]);//pushes the right parens
+		// 			}
+		// 		}
 		if (nonTerminal) {
 			if (visible) {
-				parTree.push(parens[node.cat][0]);//pushes the right parens
+				if (node["func"] && node["silentHead"]){
+					console.log("both");
```

Comment:
Please take out these "console.log"s and probably also the large commented sections above before the final merge. Thanks.